### PR TITLE
[#https://github.com/eclipse/xtext-xtend/issues/789] Improve Xbase UI

### PR DESCRIPTION
**Pull request by miklossy**
- Modify the Xbase launching infrastrukture to provide possibilities to
replace selection on an IPackageFragment element using the
JavaElementDelegate.getJavaElementForPackage(IPackageFragment) method.

Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/launching/JavaElementDelegate.java [hunk 1]
```diff
@@ -224,6 +234,14 @@ protected IJavaElement getJavaElementForResource(IResource resource) {
 		}
 		return null;
 	}
+	
+	/**
+	 * Subclasses may override.
+	 * @since 2.21 
+	 */
+	protected IJavaElement getJavaElementForPackage(IPackageFragment packageFragment) {
+		return null;
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/launching/JavaElementDelegate.java]: Why does this return null?

Comment (miklossy) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/launching/JavaElementDelegate.java]: I have chosen `null` as the return value to indicate that the Xbase framework does nothing by default and to emphasize that the subclasses should override that method to provide a meaningful return value.

**Comment by cdietrich** [comment]
@miklossy @szarnekow  how to proceed here?

**Comment by miklossy** [comment]
I would like to merge this as soon as possible and collect feedback from the community afterwards.

**Pull request closed without merging**